### PR TITLE
fix ssr modal navigator

### LINKED ISSUE #1079: 去掉 navigator.userAgent 依赖以支持服务端渲染
原计划采用类似 #1077 中的办法，增加 userAgent 属性，但是 ListItem Modal Switch 这三个组件中已经有一个 platform 属性了。只是在 Modal 里用正则判断 `isIPhone` 的时候条件不够严谨。

[Modal.web.tsx#L117](https://github.com/ant-design/ant-design-mobile/blob/241e8f2a39880eb08c67acc9d83eec1ca30f1edd/components/modal/Modal.web.tsx#L117)

```js
const isIPhone = new RegExp('\\biPhone\\b|\\biPod\\b', 'i').test(window.navigator.userAgent);
// 此处是否可以改成
const isIPhone = platform === 'ios' || (platform === 'cross' && !!navigator.userAgent.match(/\b(iPhone|iPod)\b/i));
```

这种写法有个问题就是用户必须严肃对待 platform 属性，如果在 Android 机型上故意设置 `platform='ios'` 会触发接下来的一段 hacker 代码 `{ onTouchStart: e => this.isInModal(e) }`

这里的 `new RegExp` 和之前的写法不统一，不过 `/regexp/i.test(string)` 比 `!!string.match(/regexp/i)` 更简洁。

另外 platform 的值是不是只有 'android' 'ios' 'cross' 这三种？因为代码中从没出现过 'ios'，如果是的话 `PropTypes.tsx` 里也应该改成 `platform?:  'android'|'ios'|'cross';`

#### Environment(required) | 环境（必填）

- antd-mobile version（antd-mobile版本）: 1.0.7
- Browser (or mark react-native) and its version（浏览器或react-native版本）: node
- Operating environment (e.g. OS name) and its version（操作系统版本或设备型号）:

#### What did you do? Please provide steps to re-produce your problem.（请提供复现步骤）

- List.Item
- Modal
- Switch

#### What do you expected?（预期的正常效果）

去掉 navigator.userAgent 依赖

### PR DESCRIPTION
**Pull request by cncolder**
move navigator variable outside of render

fix #1079 

* [x] Make sure you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: components/modal/Modal.web.tsx [hunk 1]
```diff
@@ -20,6 +20,10 @@ export default class Modal extends React.Component<ModalProps, any> {
   };
 
   isInModal(e) {
+    if (!/ip[honead]+/.test(navigator.userAgent)) {
```
Comment (paranoidjk) [inline components/modal/Modal.web.tsx]: 这个正则有问题

Comment (cncolder) [inline components/modal/Modal.web.tsx]: 不好意思，忘了参数 i

`/ip[honead]+/i` 这样连同 iPad 的写法可行吗？还是改回传统的 `/\b(iphone|ipod)\b/i` ？

Comment (paranoidjk) [inline components/modal/Modal.web.tsx]: /ip[honead]+/i 这个正则本身就有问题吧，中括号是之一

Comment (cncolder) [inline components/modal/Modal.web.tsx]: 忘了在哪里看到这种写法了，要是觉得不可靠我改回之前那样 `/\biPhone\b|\biPod\b/i`

Comment (paranoidjk) [inline components/modal/Modal.web.tsx]: http://regexper.com/

**Comment by paranoidjk** [comment]
@silentcloud 抽空review下这个，没问题的话合掉。

**Pull request merged**